lib/cli/ui/color.rb

```diff
@@ -20,30 +20,49 @@ def initialize(sgr, name)
         @name = name
       end
 
-      RED     = new('31', :red)
-      GREEN   = new('32', :green)
-      YELLOW  = new('33', :yellow)
+      BLACK     = new('90', :black)
+      RED       = new('31', :red)
+      GREEN     = new('32', :green)
+      YELLOW    = new('33', :yellow)
       # default blue is low-contrast against black in some default terminal color scheme
-      BLUE    = new('94', :blue) # 9x = high-intensity fg color x
-      MAGENTA = new('35', :magenta)
-      CYAN    = new('36', :cyan)
-      RESET   = new('0',  :reset)
-      BOLD    = new('1',  :bold)
-      WHITE   = new('97', :white)
+      BLUE      = new('94', :blue) # 9x = high-intensity fg color x
+      MAGENTA   = new('35', :magenta)
+      CYAN      = new('36', :cyan)
+      WHITE     = new('97', :white)
+      RESET     = new('0',  :reset)
+      BOLD      = new('1',  :bold)
+      ITALIC    = new('2',  :italic)
+      UNDERLINE = new('3',  :underline)
```

Comment:
This class is called `Color`. It is for the color portion of the ANSI SGR system. I'd be more than happy to accept this PR with modifications. The additional colors are a welcome addition.

Reply:
But Formatter is responsible for converting `{{ ... }}` syntax, the actual ANSI codes are in the `color` class and are used by other parts like the widgets. Should the `Widgets` class use `Formatter::SUCCESS` or `Color::SUCCESS` ? Is it the strict naming and the use of the colloquial `Color` that is causing you concern? Would it be more appealing if we just rename `Color` to `ANSI_SGR` instead?

Reply:
With reflection, it's definitely the name `Color` that really gives me pause here. Calling it `SGR` would alleviate that, but I agree that it's not an ideal name for understandability. Perhaps it could be renamed to `Format`, which works semantically and also works as being largely a tool of the `Formatter` class. I'd definitely be open to it if you want to take on that refactor.

Reply:
It is a bit weird to have a `CLI::UI::Format` and a similarly named `CLI::UI::Formatter`.  A few other options:
* `ANSI` (already exists, appears the author was trying to split out the roles by creating `Color` in the first place. We would merge the constants to this Class)
* `EscapeCodes` (Feels like a natural abstraction, but probably not as obvious for non-ANSI aware users.)
* `FormatCodes` (hybrid of Format and Escape Codes)